File: modules/my-module/ios/MyModule.swift
import ExpoModulesCore

public class MyModule: Module {
  public func definition() -> ModuleDefinition {
    Name("MyModule")

    View(MyModuleView.self) {
      Prop("color") { (view: MyModuleView, colorHex: String) in
        view.setInkColor(hex: colorHex)
      }
      
      Prop("isEraser") { (view: MyModuleView, eraser: Bool) in
        view.setIsEraser(eraser: eraser)
      }
      
      AsyncFunction("undo") { (view: MyModuleView) in
        view.undo()
      }
      
      Prop("strokeWidth") { (view: MyModuleView, width: Double) in
        view.setStrokeWidth(width: width)
      }
      
      Prop("penType") { (view: MyModuleView, type: String) in
        view.setPenType(type: type)
      }
      
      AsyncFunction("clear") { (view: MyModuleView) in
        view.clear()
      }
      
      AsyncFunction("saveAsImage") { (view: MyModuleView) -> String? in
        return view.saveAsImage()
      }
    }
  }
}

File: modules/my-module/android/src/main/java/expo/modules/mymodule/MyModule.kt
package expo.modules.mymodule

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class MyModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("MyModule")

    View(MyModuleView::class) {
      Prop("color") { view: MyModuleView, colorHex: String ->
        view.setInkColor(colorHex)
      }
      
      Prop("isEraser") { view: MyModuleView, eraser: Boolean ->
        view.setIsEraser(eraser)
      }
      
      AsyncFunction("undo") { view: MyModuleView ->
        view.undo()
      }

      Prop("strokeWidth") { view: MyModuleView, width: Double ->
        view.setStrokeWidth(width.toFloat())
      }
      
      Prop("penType") { view: MyModuleView, type: String ->
        view.setPenType(type)
      }
      
      AsyncFunction("clear") { view: MyModuleView ->
        view.clear()
      }
      
      AsyncFunction("saveAsImage") { view: MyModuleView ->
        return@AsyncFunction view.saveAsImage()
      }
    }
  }
}

